### ISSUE TITLE
api route compilation stack overflow error

### ISSUE THREAD

**Issue by jtmarmon** [metosin/compojure-api#123]
[call stack](https://gist.github.com/jtmarmon/cc2b2079cae61e2fd985)

this seems to happen specifically with datomic queries within a let statement. when running `defapi` on one of those routes (i.e. no error when compiling just `defroutes*`). trying to dig up more info here...

**Comment by jtmarmon** [comment]
fyi this has happened twice with two completely distinct routes

**Comment by jtmarmon** [comment]
alright figured it out: if you have a let statement that overrides the name of one of your `defroutes*`, it will recur infinitely.

e.g.:

```clj
(defroutes* foo
   (GET* "/bar" []
     (let [foo 5] 
       foo)))
```

this will throw an error when compiled via `defapi`

**Comment by ikitommi** [comment]
that's bad. we'll try to figure out some way to fix this. PR would be welcome too.

**Comment by ikitommi** [comment]
created a branch with a failing test. https://github.com/metosin/compojure-api/tree/api_route_compilation_stack_overflow_error

**Comment by ikitommi** [comment]
I read the Clojure Reader docs and I think I understand the root cause. The actual code works ok, but the source code stored as Var's metadata is broken due to use of qualified names with local symbols. 

So, in the source code analysis phase (to generate the static route tree for swagger docs), the meta-data from the original Var is seen also with the local symbols.

Have some ideas how to fix this.

**Comment by ikitommi** [comment]
http://clojure.org/reader:

```
For Symbols, syntax-quote resolves the symbol in the current context, yielding a fully-qualified symbol (i.e. namespace/name or fully.qualified.Classname). If a symbol is non-namespace-qualified and ends with '#', it is resolved to a generated symbol with the same name to which '_' and a unique id have been appended. e.g. x# will resolve to x_123. All references to that symbol within a syntax-quoted expression resolve to the same generated symbol.
```

**Comment by ikitommi** [comment]
Should be fixed in the  `[0.21.0-SNAPSHOT]`.
